Content.Server/_CP14/Sponsor/CP14SponsorManager.cs

```diff
@@ -0,0 +1,130 @@
+    [Dependency] private readonly DiscordAuthManager _discordAuthManager = default!;
+    [Dependency] private readonly INetManager _netMgr = default!;
+    [Dependency] private readonly IPrototypeManager _prototype = default!;
+
+    private readonly HttpClient _httpClient = new();
+    private string _apiUrl = string.Empty;
+    private string _apiKey = string.Empty;
+    private bool _enabled;
+
+    private ISawmill _sawmill = null!;
+
+    private Dictionary<NetUserId, HashSet<CP14SponsorRolePrototype>> _cachedRoles = new();
+
+    public void Initialize()
+    {
+        _sawmill = Logger.GetSawmill("sponsors");
+
+        _cfg.OnValueChanged(CCVars.DiscordAuthEnabled, val => { _enabled = val; }, true);
+        _cfg.OnValueChanged(CCVars.DiscordAuthUrl, val => { _apiUrl = val; }, true);
+        _cfg.OnValueChanged(CCVars.DiscordAuthToken, val => { _apiKey = val; }, true);
+
+        _discordAuthManager.PlayerVerified += OnPlayerVerified;
+        _netMgr.Disconnect += OnDisconnect;
+
+        _httpClient.DefaultRequestHeaders.Authorization =
+            new AuthenticationHeaderValue("Bearer", _apiKey);
+    }
+
+    private async Task<List<string>?> GetRoles(NetUserId userId)
+    {
+        var requestUrl = $"{_apiUrl}/roles?method=uid&id={userId}&guildId={DiscordAuthManager.DISCORD_GUILD}";
+        var response = await _httpClient.GetAsync(requestUrl);
+
+        if (!response.IsSuccessStatusCode)
+        {
+            _sawmill.Error($"Failed to retrieve roles for user {userId}: {response.StatusCode}");
+            return null;
+        }
+
+        var responseContent = await response.Content.ReadFromJsonAsync<RolesResponse>();
+
+        if (responseContent is not null)
+        {
+            return responseContent.Roles.ToList();
+        }
+
+        _sawmill.Error($"Roles not found in response for user {userId}");
+        return null;
+    }
+
+    private async void OnPlayerVerified(object? sender, ICommonSession e)
+    {
+        if (!_enabled)
+            return;
+
+        var roles = await GetRoles(e.UserId);
+        if (roles is null)
+            return;
+
+        foreach (var role in _prototype.EnumeratePrototypes<CP14SponsorRolePrototype>())
+        {
+            if (!roles.Contains(role.DiscordRoleId))
+                continue;
+
+            if (!_cachedRoles.TryGetValue(e.UserId, out var value))
+            {
+                value = new HashSet<CP14SponsorRolePrototype>();
+                _cachedRoles[e.UserId] = value;
+            }
+
+            value.Add(role);
+        }
+    }
+
+    private void OnDisconnect(object? sender, NetDisconnectedArgs e)
+    {
+        //Remove cached roles
+        if (_cachedRoles.ContainsKey(e.Channel.UserId))
+        {
+            _cachedRoles.Remove(e.Channel.UserId);
+        }
+    }
+
+    public bool UserHasFeature(NetUserId userId, ProtoId<CP14SponsorFeaturePrototype> feature)
+    {
+        if (!_enabled)
+            return false;
```

Comment:
```suggestion
            return true;
```

Если система спонсорки отключена, все фичи должны быть доступны всем